app.py

```diff
@@ -0,0 +1,139 @@
+import json
+import re
+from datetime import datetime as dt
+from pprint import pprint
+from package.broom import Player, Deck, Game, make_deck
+
+
+app = Flask(__name__)
+CORS(app, resources={r"/makedeck": {"origins": "http://localhost:8080"}})
+# CORS(app, resources={r"/makedeck": {"origins": "http://192.168.1.106:8080"}})
+
+app.config['CORS_HEADERS'] = 'Content-Type'
+
+client_sessions = {}
+
+class ClientSession:
+    
+    id_count = 0
+
+    def __init__(self, player1, player2, deck, game):
+        ClientSession.id_count += 1
+        self.id = ClientSession.id_count
+        self.p1 = player1
+        self.p2 = player2
+        self.deck = deck
+        self.g = game
+
+@app.route("/makedeck", methods=["GET", "POST"])
+@cross_origin(origin='localhost',headers=['Content- Type','Authorization'])
+def makedeck():
+
+    # TODO new random seed deck
+
+    # response.headers.add('Access-Control-Allow-Origin', '*')
+
+    if request.method == "POST":
+        context = request.get_json(force=True)
+        csId = context['clientSessionId']
+        cs = client_sessions[csId]
+        response = jsonify({
+            "game_state": cs.g.returnJSON(),
+            "id": cs.id,
+            'post_response': True
+        })
+
+        return response
+    else:
+        p1 = Player('player_1', 0)
+        p2 = Player('player_2', 0)
+        deck = Deck(make_deck())
+        g = Game(p1,p2,deck)
+
+        cs = ClientSession(p1, p2, deck, g)
+        client_sessions[cs.id] = cs
+        response = jsonify({
+            "game_state": g.returnJSON(),
+            "id": cs.id
+        })
+
+        return response
+
+@app.route("/playfirstround", methods=["GET", "POST"])
+@cross_origin(origin='localhost',headers=['Content- Type','Authorization'])
+def playfirstround():
+
+    items = {
+    }
+    response = jsonify(items)
+    # response.headers.add('Access-Control-Allow-Origin', '*')
+
+    if request.method == "POST":
+
+        context = request.get_json(force=True)
+        csId = context['clientSessionId']
+        cs = client_sessions[csId]
+        game = cs.g.play_first_round(cs.p1, cs.p2).returnJSON()
+        #game = game.returnJSON()
+        return jsonify({
+            "game_state": game,
+            "id": cs.id
+        })
+
+        #response = g.play_first_round(p1, p2).returnJSON()
+        # round_results = f"Player 1 score: {p1.score}\n\tPlayer 2 score: {p2.score}"
+
+    else:
+        return response
+
+@app.route("/validplays", methods=["GET", "POST"])
+@cross_origin(origin='localhost',headers=['Content- Type','Authorization'])
+def validplays():
+    
+    if request.method == "POST":
+
+        context = request.get_json(force=True)
+        csId = context['clientSessionId']
+        cs = client_sessions[csId]
+
+        first_player = list(cs.g.valid_plays(cs.p1, cs.g.table_cards))
+        second_player = list(cs.g.valid_plays(cs.p1, cs.g.table_cards))
+
+        return jsonify({
```

Comment:
post client session id

Reply:
so it's about being idiomatic. GET is used when asking something that the server knows or can do. POST is used when the user give new info that the server couldn't have (like a a comment, or filling in a form, etc)

A GET can be used with a json as an "argument" with semantics as follows (from the perspective of the client-side): Hey backend, given this <int or string, or list of things, or json> tell me what I want to know about <whatever the endpoint does>

A POST is more like: Hey backend, here is a <int or string, or list of things or json> to put somewhere, tell me that you were successful at adding it to your records.

The key difference is in the second half of both statements. For the GET the response was info/action by the server. The POST the response was confirmation of the addition (this can be more than a bool, like the length of bits added, the length of the new list of things, a blob representing the collection that has been modified, etc). This is crucial for creating api's that are intuitive for others to use. You may be wondering, "why the big deal, it's all just request/response anyway". It's about giving the users/consumers of your api's an intuition about how they work. In a nutshell, GET shouldn't modify the backend - POST may/should. This isn't a hard rule.

In the case of "clientSessionID", depending on when/where this unique id is generated, it may have to be "posted" once (if generated by a client, the server needs to know about it). Afterwards, it can simply put it as a param to it's GETs and the server can use it to look up the appropriate info. But that shouldn't make the "/validplays" endpoint POST every time - that logic belongs under GET.